### ISSUE TITLE
Runtime manifest doesn't respect [contenthash]

### ISSUE THREAD

**Issue by tselishev-semen** [GoogleChrome/workbox#2021]
**Library Affected**:
workbox-webpack-plugin@4.2.0

**Browser & Platform**:
all browsers

**Issue or Feature Request Description**:
Webpack provides ability to use [contenthash] in outputnames (see https://webpack.js.org/guides/caching/) In this case, when I'm using GenerateSW from workbox-webpack-plugin library, my generated precache-manifest.js doesn't respect this option and generates it's own revision number, after that the resources could be downloaded from the network two times,  if they are in the critical rendering path for the app and in the manifest.

```
self.__precacheManifest = (self.__precacheManifest || []).concat([

  {
    "revision": "f62b7714d7f5a2ad8629",
    "url": "/resources/publisher.c537d714c8ba995c06cc.css"
  },
]);

**Comment by jeffposnick** [comment]
C.f. https://github.com/webpack/webpack/issues/9038

**Comment by jeffposnick** [comment]
Inferring this automatically isn't going to happen soon, unfortunately.

In v5, we're going to support the `dontCacheBustURLsMatching` config option for this, and you'd use it along the lines of

```js
WorkboxPlugin({
  dontCacheBustURLsMatching: /\.[0-9a-f]{20}\./,
  // Other Workbox config.
})
```

(Changing the `RegExp` to match different patterns depending on the length of your hashes and naming convention.)

**Comment by jeffposnick** [comment]
This should be addressed by the current Workbox v5.0.0 alpha.
